build-tools/packages/build-cli/test/library/packageExports.test.ts

```diff
@@ -0,0 +1,331 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "chai";
+
+import type { ExportData, Node10CompatExportData } from "../../src/library/packageExports";
+import { queryResolutionPathsFromPackageExports } from "../../src/library/packageExports";
+
+import type { Logger, PackageJson } from "@fluidframework/build-tools";
+
+const typeOnlyExportPackage: PackageJson = {
+	name: "@fluid-test/test-package-typeonly",
+	scripts: {},
+	version: "",
+	exports: {
+		".": {
+			import: {
+				types: "./lib/public.d.ts",
+			},
+			require: {
+				types: "./dist/public.d.ts",
+			},
+		},
+		"./alpha": {
+			types: {
+				// a bit unusual authoring where any non-require types lookup
+				// will get resolved. (explicit import condition is common.)
+				require: "./dist/alpha.d.ts",
+				default: "./lib/alpha.d.ts",
+			},
+		},
```

Comment:
The different structure here compared to all the other test data is to make sure we have a test for it just in case? Or do we actually have a package that does things differently like this?

Reply:
We do not have any case of this that I know of. This is a legitimate structuring; so, gave it a little coverage.